framework/deproxy_client.py

```diff
@@ -688,7 +687,7 @@ def handle_write(self):
 
     @staticmethod
     def __headers_to_string(headers):
-        return "".join(["%s: %s\r\n" % (h, v) for h, v in headers])
+        return "".join(["%s: %s\r\n" % (h.decode("utf-8"), v.decode("utf-8")) for h, v in headers])
```

Comment:
You should use `__binary_headers_to_string` for this. I think this method is unnecessary. Please remove it.

Reply:
okay.